### ISSUE TITLE
Merge with pyxboxlive project

### ISSUE THREAD

**Issue by titilambert** [tuxuser/py-xbox-webapi#3]
Hello, I just started a new python 3 xboxlive api, using asyncio: https://github.com/titilambert/pyxboxlive
The authentication part of pyxboxlive is not really nice ...
So, do you want to merge your project with mine ? Or do you want merge my project with yours ? 

Thanks !

**Comment by tuxuser** [comment]
Hey, thanks for the offer!

IMHO we should work together, there are too many libs already which just just have a fraction of API endpoints implemented.

My choice for requests as http lib was simple: It's the most comfortable python lib ever written...and its widely used, everybody has got it installed.

I would appreciate to have you on the "py-xbox-webapi" team.

**Comment by titilambert** [comment]
The request choice is completly normal. This is kind of standard. On my side, I choose aiohttp to be able to use it with asyncio. I also choose click to create a command line easily. My final objective is to create a home-assistant (http://home-assistant.io) component to be able to see which user is connected on the xbox one and which application is running...

**Comment by tuxuser** [comment]
Do you need help with a specific API endpoint?

**Comment by titilambert** [comment]
Not really, all endpoinds implemented are working well. The main issue is about authentication stability.
Also, I'm looking for endpoints to control remotely the xbox one ( volume, play/pause, current channel, start apps, shutdown, ...)

**Comment by tuxuser** [comment]
Yeah, I admit .. The Authentication manager is more complicated than it should be - other possible issue is that my version requires userinput via stdin 

For remote controlling another protocol is useful to look at, it's smartglass. Check out @joelday 's Github.

**Comment by titilambert** [comment]
@tuxuser thanks for your answer.
BTW what is your goal with this library ?

**Comment by tuxuser** [comment]
Tbh, it already reached it's goal - it can authenticate with Xbox Live... And supports Two-Factor-Authentication.
That's all I wanted. People can grab their token and use it on all the endpoints they can find.

Three points I would happily accept Pull Requests for tho:
1) Working implementation of [Device Authentication](https://github.com/tuxuser/py-xbox-webapi/blob/master/xbox_webapi/authentication/auth.py#L524) and [Title Authentication](https://github.com/tuxuser/py-xbox-webapi/blob/master/xbox_webapi/authentication/auth.py#L496) - Not sure what it's used for lol, but nice to have ^^
2) Implementation of additional API endpoints
3) Fixes

**Comment by titilambert** [comment]
Oki !
So, I have some questions: 
- Are you open to use asyncio and aiohttp instead of request ? This is the new standard with python 3.
- Are you open to add a CLI to perform some command with a terminal ?
- What is device/title authentication ?
- Where do you find those endpoints and how to use is ?

Thanks !

**Comment by tuxuser** [comment]
1) I would prefer making the Backend pluggable, so you can choose if aiohttp or requests is used ;)
2) Sure, that could be added - The API should be designed as comfortable as a CLI ideally
3) Device and Title Auth is normally just done by the real Xbox console, not by a PC application. It's very unlikely that will achieve that - and I really dont know what it is used for.
4) The endpoints are found by looking at the official Xbox API codebase (https://github.com/Microsoft/Xbox-live-api )

**Comment by titilambert** [comment]
I have an issue with the point 1.
We can not do that, Because of Python, we have to chose "async" or "sync".
So, first question: Do you know the differences between "async" or "sync" methods ?
If not, please check: https://docs.python.org/3/library/asyncio.html
Then, what do you think about using async instead of sync ?
